Framework/CurveFitting/src/Algorithms/LeBailFit.cpp

```diff
@@ -471,31 +472,34 @@ void LeBailFit::execPatternCalculation() {
   // Calculate peak intensities and diffraction pattern
   vector<double> emptyvec;
   bool resultphysical = calculateDiffractionPattern(
-      m_dataWS->readX(m_wsIndex), m_dataWS->readY(m_wsIndex), true, true,
-      emptyvec, vecY, rfactor);
+      m_dataWS->x(m_wsIndex).rawData(), m_dataWS->y(m_wsIndex).rawData(), true,
+      true, emptyvec, vecY, rfactor);
+  m_outputWS->mutableY(CALDATAINDEX) = vecY;
 
   size_t numpts = vecY.size();
   // Calculate background
-  MantidVec &vec_bkgd = m_outputWS->dataY(INPUTBKGDINDEX);
+  std::vector<double> vec_bkgd(m_outputWS->y(INPUTBKGDINDEX).size(), 0);
   m_lebailFunction->function(vec_bkgd, vecX, false, true);
+  m_outputWS->mutableY(INPUTBKGDINDEX) = vec_bkgd;
+
   for (size_t i = 0; i < numpts; ++i)
-    m_outputWS->dataY(INPUTPUREPEAKINDEX)[i] =
-        m_outputWS->readY(OBSDATAINDEX)[i] -
-        m_outputWS->readY(INPUTBKGDINDEX)[i];
+    m_outputWS->mutableY(INPUTPUREPEAKINDEX)[i] =
+        m_outputWS->y(OBSDATAINDEX)[i] - m_outputWS->y(INPUTBKGDINDEX)[i];
```

Comment:
I think you can just write `m_outputWS->mutableY(INPUTPUREPEAKINDEX) = m_outputWS->y(OBSDATAINDEX) - m_outputWS->y(INPUTBKGDINDEX)`.